### ISSUE TITLE
Curses::addch doesn't support attributes

### ISSUE THREAD

**Issue by coezbek** [ruby/curses#59]
The documentation says that `addch` would support outputting characters with attributes, but in practice it doesn't work.

Ways to reproduce:

```ruby
require "curses"
include Curses

init_screen
begin
  addstr("The following letter A should be BOLD and UNDERLINED by using addch:\n")
  addch('A'.ord | A_BOLD | A_UNDERLINE)

  addstr("\nIt should look the same as when using attron and addstr:\n")
  attron(A_BOLD | A_UNDERLINE)
  addstr("A")
  getch
ensure
  close_screen
end
```

What I am seeing:

![image](https://user-images.githubusercontent.com/12127567/117054769-c0c7b300-ad1a-11eb-8ca5-7fb69905ac8d.png)

Is just the documentation inaccurate or does the conversion from NUM2CH remove the attributes again?

https://github.com/ruby/curses/blob/3e97c295fc2f5348deeec90e821360272ed2f423/ext/curses/curses.c#L765

**Comment by shugo** [comment]
I've fixed in 5e45ef9.  Please try it.

**Comment by coezbek** [comment]
Did try it and can confirm that 1.4.1 fixes this issue. Thank you @shugo.
